packages/editor/src/components/template-validation-notice/index.js

```diff
@@ -1,63 +1,62 @@
 /**
  * WordPress dependencies
  */
-import { Notice } from '@wordpress/components';
+import {
+	Notice,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 
-function TemplateValidationNotice( { isValid, ...props } ) {
+export default function TemplateValidationNotice() {
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+	const { isValid } = useSelect( ( select ) => {
+		return {
+			isValid: select( blockEditorStore ).isValidTemplate(),
+		};
+	}, [] );
+	const { setTemplateValidity, synchronizeTemplate } =
+		useDispatch( blockEditorStore );
+
 	if ( isValid ) {
 		return null;
 	}
 
-	const confirmSynchronization = () => {
-		if (
-			// eslint-disable-next-line no-alert
-			window.confirm(
-				__(
-					'Resetting the template may result in loss of content, do you want to continue?'
-				)
-			)
-		) {
-			props.synchronizeTemplate();
-		}
-	};
-
 	return (
-		<Notice
-			className="editor-template-validation-notice"
-			isDismissible={ false }
-			status="warning"
-			actions={ [
-				{
-					label: __( 'Keep it as is' ),
-					onClick: props.resetTemplateValidity,
-				},
-				{
-					label: __( 'Reset the template' ),
-					onClick: confirmSynchronization,
-				},
-			] }
-		>
-			{ __(
-				'The content of your post doesn’t match the template assigned to your post type.'
-			) }
-		</Notice>
+		<>
+			<Notice
+				className="editor-template-validation-notice"
+				isDismissible={ false }
+				status="warning"
+				actions={ [
+					{
+						label: __( 'Keep it as is' ),
+						onClick: () => setTemplateValidity( true ),
+					},
+					{
+						label: __( 'Reset the template' ),
+						onClick: () => setShowConfirmDialog( true ),
+					},
+				] }
+			>
+				{ __(
+					'The content of your post doesn’t match the template assigned to your post type.'
+				) }
+			</Notice>
+			<ConfirmDialog
+				isOpen={ showConfirmDialog }
```

Comment:
```suggestion
				isOpen={ showConfirmDialog }
				confirmButtonText={ __( 'Reset' ) }
```

Considering #60256 and #46859, is "Reset" better than "OK"?